rpms/pulp/pulp.spec

```diff
@@ -0,0 +1,853 @@
+%{_var}/lib/%{name}/uploads
+%{_var}/www/pub
+%{_var}/www/pulp
+
+%defattr(640,apache,apache,750)
+%dir %{_var}/log/%{name}
+
+
+%pre server
+# If we are upgrading
+if [ $1 -gt 1 ] ; then
+        /bin/systemctl stop pulp_workers > /dev/null 2>&1
+        /bin/systemctl stop pulp_celerybeat > /dev/null 2>&1
+        /bin/systemctl stop pulp_resource_manager > /dev/null 2>&1
+fi
+
+
+%post server
+# RSA key pair
+KEY_DIR="%{_sysconfdir}/pki/%{name}"
+KEY_PATH="$KEY_DIR/rsa.key"
+KEY_PATH_PUB="$KEY_DIR/rsa_pub.key"
+if [ ! -f $KEY_PATH ]
+then
+  openssl genrsa -out $KEY_PATH 2048 &> /dev/null
+  openssl rsa -in $KEY_PATH -pubout > $KEY_PATH_PUB 2> /dev/null
+fi
+
+# CA certificate
+if [ $1 -eq 1 ]; # not an upgrade
+then
+  pulp-gen-ca-certificate
+fi
+
+%systemd_post pulp_workers.service
+%systemd_post pulp_celerybeat.service
+%systemd_post pulp_resource_manager.service
+%systemd_post pulp_streamer.service
+
+
+%preun server
+%systemd_preun pulp_workers.service
+%systemd_preun pulp_celerybeat.service
+%systemd_preun pulp_resource_manager.service
+%systemd_preun pulp_streamer.service
+
+
+%postun server
+%systemd_postun
+
+
+# ---- Agent Handler Framework -------------------------------------------------
+%package -n python2-pulp-agent-lib
+Summary: Pulp agent handler framework
+Requires: python2-%{name}-common = %{version}
+%{?python_provide:%python_provide python2-pulp-agent-lib}
+
+
+%description -n python2-pulp-agent-lib
+A framework for loading agent handlers that provide support
+for content, bind and system specific operations.
+
+
+%files -n python2-pulp-agent-lib
+%license LICENSE
+%dir %{_sysconfdir}/%{name}/agent
+%dir %{_sysconfdir}/%{name}/agent/conf.d
+%dir %{_usr}/lib/%{name}/agent
+%{python2_sitelib}/%{name}/agent/
+%{python2_sitelib}/pulp_agent*.egg-info
+%doc README COPYRIGHT
+
+
+# ---- Client Bindings ---------------------------------------------------------
+%package -n python2-pulp-bindings
+Summary: Pulp REST bindings for python
+Requires: m2crypto
+Requires: python2-oauth2 >= 1.5.170-2.pulp
+Requires: python2-%{name}-common = %{version}
+%{?python_provide:%python_provide python2-pulp-bindings}
+
+
+%description -n python2-pulp-bindings
+The Pulp REST API bindings for python.
+
+
+%files -n python2-pulp-bindings
+%license LICENSE
+%{python2_sitelib}/%{name}/bindings/
+%{python2_sitelib}/pulp_bindings*.egg-info
+%doc README COPYRIGHT
+
+
+# ---- Client Extension Framework -----------------------------------------------------
+%package -n python2-pulp-client-lib
+Summary: Pulp client extensions framework
+Requires: m2crypto
+Requires: python-isodate >= 0.5.0-1
+Requires: python-okaara >= 1.0.32
+Requires: python2-%{name}-common = %{version}
+Requires: python2-setuptools
+%{?python_provide:%python_provide python2-client-lib}
+
+
+%description -n python2-pulp-client-lib
+A framework for loading Pulp client extensions.
+
+
+%files -n python2-pulp-client-lib
+%license LICENSE
+%dir %{python2_sitelib}/%{name}/client
```

Comment:
If the package owns a dir, does it not automatically own all the files within the dir?